core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -204,21 +209,22 @@ public void setTransportSupplier(Supplier<ClientTransport> supplier) {
       if (shutdown && pendingStreams != null) {
         listener.transportTerminated();
       }
-      savedPendingStreams = pendingStreams;
+      // createRealStream may be expensive. It will start real streams on the transport. If there
+      // are pending requests, they will be serialized too, which may be expensive. Since we are
+      // now on transport thread, we need to offload the work to an executor.
+      if (pendingStreams != null) {
+        for (final PendingStream stream : pendingStreams) {
+          streamCreationExecutor.execute(new Runnable() {
```

Comment:
This could cause a burst causing many thread creations. Are we concerned about that?